tests/testthat/test-developer-expected_event.R

```diff
@@ -28,3 +28,65 @@ test_that("expected event vs gsDesign", {
   expect_equal(x1, x2)
   expect_equal(x2, x3)
 })
+
+test_that("expected_event returns consistent results (regression tests)", {
+  # Default arguments, simple output (total event count only)
+  observed <- expected_event()
+  expected <- 57.3537033783035
```

Comment:
In commit 4bd45fb15bc9bf6b326f4b29f13206be60e17164, I added a note at the top of the regression tests to explain the code version of gsDesign2 that I used to generate the expected results.

In more practical terms, I ran each `expected_event()`  example call and then ran `dput()` on its output to construct the values in `expected`

Reply:
Since `v1.1.0` is not available as CRAN or GitHub release, shall we call the commit ID rather than the version number?

Reply:
I'm not sure what you would like me to change. The commit ID is the last part of the URL I included `https://github.com/Merck/gsDesign2/commit/3406261092edcb696cecd0392ff94e08e832971a`. And I mentioned the version number for more context. In other words, a year from now, just the date or commit ID won't be as immediately interpretable. But if a year from now the version is eg `3.0.1`, you would know that the expected results were computed two major versions ago.

The note currently reads:

```
  # Note: all expected outputs were computed on 2023-08-09 with gsDesign2
  # version 1.1.0 and specifically the package was installed from GitHub from
  # the following commit
  # https://github.com/Merck/gsDesign2/commit/3406261092edcb696cecd0392ff94e08e832971a
```

If you would like to suggest an alternative, I'm happy to replace this text

Reply:
> My concern is the version you are developing is v1.1.0. If you compare your version with v1.1.0, then it is the comparison between 1.1.0 and 1.1.0. This might be confusing to users.

Ah, ok. Now I understand. You make a good point. Version 1.1.0 is a moving target, so it is not a good candidate for producing the expected results for regression tests.

However, I confirmed that the results are identical with the latest CRAN version 1.0.9. The only required change is replacing the new `define_*` functions with their equivalent `tibble()` calls.

``` r
library(tibble)
library(gsDesign2)
packageVersion("gsDesign2")
#> [1] '1.0.9'

# Default arguments, simple output (total event count only)
expected <- expected_event()
dput(expected)
#> 57.3537033783035

# Event count by time period
expected <- expected_event(simple = FALSE)
dput(expected)
#> structure(list(t = c(0, 3), fail_rate = c(0.0770163533955495, 
#> 0.0385081766977747), event = c(22.2482399817186, 35.1054633965849
#> )), row.names = c(NA, -2L), class = c("tbl_df", "tbl", "data.frame"
#> ))

# Early cutoff
expected <- expected_event(total_duration = .5)
dput(expected)
#> 0.0285092329152224

# Single time period example
expected <- expected_event(
  enroll_rate = tibble(duration = 10, rate = 10),
  fail_rate = tibble(duration = 100, fail_rate = log(2) / 6, dropout_rate = .01),
  total_duration = 22,
  simple = FALSE
)
dput(expected)
#> structure(list(t = 0, fail_rate = 0.115524530093324, event = 80.4097370913342), row.names = c(NA, 
#> -1L), class = c("tbl_df", "tbl", "data.frame"))

# Single time period example, multiple enrollment periods
expected <- expected_event(
  enroll_rate = tibble(duration = c(5, 5), rate = c(10, 20)),
  fail_rate = tibble(duration = 100, fail_rate = log(2) / 6, dropout_rate = .01),
  total_duration = 22, simple = FALSE
)
dput(expected)
#> structure(list(t = 0, fail_rate = 0.115524530093324, event = 118.848383110223), row.names = c(NA, 
#> -1L), class = c("tbl_df", "tbl", "data.frame"))
```

<sup>Created on 2023-08-11 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.3.1 (2023-06-16)
#>  os       Ubuntu 22.04.3 LTS
#>  system   x86_64, linux-gnu
#>  ui       X11
#>  language (EN)
#>  collate  C.UTF-8
#>  ctype    C.UTF-8
#>  tz       America/New_York
#>  date     2023-08-11
#>  pandoc   3.1.1 @ /usr/lib/rstudio/resources/app/bin/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version date (UTC) lib source
#>  cli           3.6.1   2023-03-23 [3] RSPM (R 4.2.0)
#>  colorspace    2.1-0   2023-01-23 [3] RSPM (R 4.2.0)
#>  digest        0.6.33  2023-07-07 [3] RSPM (R 4.2.0)
#>  dplyr         1.1.2   2023-04-20 [3] RSPM (R 4.2.0)
#>  evaluate      0.21    2023-05-05 [3] RSPM (R 4.2.0)
#>  fansi         1.0.4   2023-01-22 [3] RSPM (R 4.2.0)
#>  fastmap       1.1.1   2023-02-24 [3] RSPM (R 4.2.0)
#>  fs            1.6.3   2023-07-20 [3] RSPM (R 4.2.0)
#>  generics      0.1.3   2022-07-05 [3] RSPM (R 4.2.0)
#>  ggplot2       3.4.2   2023-04-03 [3] RSPM (R 4.2.0)
#>  glue          1.6.2   2022-02-24 [3] RSPM (R 4.2.0)
#>  gsDesign      3.5.0   2023-07-19 [3] RSPM (R 4.2.0)
#>  gsDesign2   * 1.0.9   2023-06-20 [3] RSPM (R 4.2.0)
#>  gt            0.9.0   2023-03-31 [3] RSPM (R 4.2.0)
#>  gtable        0.3.3   2023-03-21 [3] RSPM (R 4.2.0)
#>  htmltools     0.5.5   2023-03-23 [3] RSPM (R 4.2.0)
#>  knitr         1.43    2023-05-25 [3] RSPM (R 4.2.0)
#>  lifecycle     1.0.3   2022-10-07 [3] RSPM (R 4.2.0)
#>  magrittr      2.0.3   2022-03-30 [3] RSPM (R 4.2.0)
#>  munsell       0.5.0   2018-06-12 [3] RSPM (R 4.2.0)
#>  mvtnorm       1.2-2   2023-06-08 [3] RSPM (R 4.2.0)
#>  pillar        1.9.0   2023-03-22 [3] RSPM (R 4.2.0)
#>  pkgconfig     2.0.3   2019-09-22 [3] RSPM (R 4.2.0)
#>  purrr         1.0.1   2023-01-10 [3] RSPM (R 4.2.0)
#>  R.cache       0.16.0  2022-07-21 [3] RSPM (R 4.2.0)
#>  R.methodsS3   1.8.2   2022-06-13 [3] RSPM (R 4.2.0)
#>  R.oo          1.25.0  2022-06-12 [3] RSPM (R 4.2.0)
#>  R.utils       2.12.2  2022-11-11 [3] RSPM (R 4.2.0)
#>  R6            2.5.1   2021-08-19 [3] RSPM (R 4.2.0)
#>  Rcpp          1.0.11  2023-07-06 [3] RSPM (R 4.2.0)
#>  reprex        2.0.2   2022-08-17 [3] RSPM (R 4.2.0)
#>  rlang         1.1.1   2023-04-28 [3] RSPM (R 4.2.0)
#>  rmarkdown     2.23    2023-07-01 [3] RSPM (R 4.2.0)
#>  rstudioapi    0.15.0  2023-07-07 [3] RSPM (R 4.2.0)
#>  scales        1.2.1   2022-08-20 [3] RSPM (R 4.2.0)
#>  sessioninfo   1.2.2   2021-12-06 [3] RSPM (R 4.2.0)
#>  styler        1.10.1  2023-06-05 [3] RSPM (R 4.2.0)
#>  tibble      * 3.2.1   2023-03-20 [3] RSPM (R 4.3.0)
#>  tidyr         1.3.0   2023-01-24 [3] RSPM (R 4.2.0)
#>  tidyselect    1.2.0   2022-10-10 [3] RSPM (R 4.2.0)
#>  utf8          1.2.3   2023-01-31 [3] RSPM (R 4.2.0)
#>  vctrs         0.6.3   2023-06-14 [3] RSPM (R 4.2.0)
#>  withr         2.5.0   2022-03-03 [3] RSPM (R 4.2.0)
#>  xfun          0.39    2023-04-20 [3] RSPM (R 4.2.0)
#>  xml2          1.3.5   2023-07-06 [3] RSPM (R 4.2.0)
#>  xtable        1.8-4   2019-04-21 [3] RSPM (R 4.2.0)
#>  yaml          2.3.7   2023-01-23 [3] RSPM (R 4.2.0)
#> 
#>  [1] /home/jdblischak/R/x86_64-pc-linux-gnu-library/4.3
#>  [2] /usr/local/lib/R/site-library
#>  [3] /usr/lib/R/site-library
#>  [4] /usr/lib/R/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

Reply:
Though I can't use `tibble::tibble()` in the tests for this PR, because the latest `expected_event()` already throws a warning if not using the `define_*` functions

Reply:
Also, I forgot to run the final example with version 1.0.9. The output is also identical:

```R
expected <- expected_event(
  enroll_rate = tibble::tibble(duration = c(2, 1, 2), rate = c(5, 10, 20)),
  fail_rate = tibble::tibble(duration = c(1, 1, 100), fail_rate = c(0.05, 0.02, 0.01), dropout_rate = 0.01),
  total_duration = 12,
  simple = FALSE
)
dput(expected)
#> structure(list(t = c(0, 1, 2), fail_rate = c(0.05, 0.02, 0.01
#> ), event = c(2.91177332078756, 1.11333393252082, 3.45481304353542
#> )), row.names = c(NA, -3L), class = c("tbl_df", "tbl", "data.frame"
#> ))
```

Reply:
Thanks for the systematic comparisons. I like the comments, which are super clear.